src/Matrix.jl

```diff
@@ -2494,16 +2494,35 @@ julia> minors(A, 2)
 
 ```
 """
-function minors(A::MatElem, k::Int)
-   row_indices = combinations(nrows(A), k)
-   col_indices = combinations(ncols(A), k)
-   mins = Vector{elem_type(base_ring(A))}(undef, 0)
-   for ri in row_indices
-      for ci in col_indices
-         push!(mins, det(A[ri, ci]))
-      end
-   end
-   return(mins)
+minors(A::MatElem, k::Int) = collect(minors_iterator(A, k))
```

Comment:
Should we just replace `minors` by `minors_iterator` (i.e. rename the latter to `minors`) ? Perhaps not because it would be breaking, but other than this (clearly important) concern I see no downsides?

I don't want to hold up this PR for it, but wanted to at least mention it.

Reply:
Maybe @fieker or @thofma have opinions on this

Reply:
The docstring of minors states clearly that it returns an array, so I would consider this breaking (at least for AA).
But we could discuss if we consider this breaking for Oscar, or if it is fine to do in a breaking AA release and a minor Oscar release afterwards

Reply:
It definitely would be breaking, regardless of what the docstring says, because there could be code which assumes it is an array and accesses it as such.

And in fact Oscar has a lot of code looking like this:
```julia
  R, (x, y, z, w) = QQ[:x, :y, :z, :w]

  M = R[x y z; y-1 z-2 w]

  I = ideal(R, minors(M, 2))
```

So I think we should leave it as is (and only consider changing it for "Oscar 2.0" if at all)

Reply:
Thanks everyone for looking at this so quickly! I agree that one should be very careful about changing minors to return an iterator, even though I also agree that that would make more sense in the long term. 

As for the point that @fingolfin raises above: Perhaps it would make sense to allow users to construct ideals from generators that come from an iterator, rather than just vectors?